### PR TITLE
Add Secrets Sync documentation

### PR DESCRIPTION
**Pull request by vinay-gopalan**
Adds documentation for the TFVP Secrets Sync resources. Includes:
- Destination documentation (AWS, GCP, Azure, GitHub, Vercel)
- Association documentation

### CODE REVIEW

**Review by raymonstah — APPROVED** [review]

File: website/docs/r/secrets_sync_association.html.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    {
+      dev  = "B!gS3cr3t",
+      prod = "S3cureP4$$"
+    }
+  )
+}
+
+resource "vault_secrets_sync_gh_destination" "gh" {
+  name = "gh-dest"
+}
```
Comment (maxcoulombe) [inline website/docs/r/secrets_sync_association.html.md:47]: Should the example use explicit token, repo & owner fields?

File: website/docs/r/secrets_sync_aws_destination.html.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+## Example Usage
+
+```hcl
+resource "vault_secrets_sync_aws_destination" "aws" {
+  name              = "aws-dest"
+  access_key_id     = var.access_key_id
+  secret_access_key = var.secret_access_key
+  region            = "us-east-1"
+}
+```
```
Comment (maxcoulombe) [inline website/docs/r/secrets_sync_aws_destination.html.md:33]: Should we give an example for the custom_tags & secret_name_template on each destination where applicable? Consistency is always useful, we should decide to either use all optional customisable fields or none in the example usage for all destinations.

File: website/docs/r/secrets_sync_azure_destination.html.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+* `name` - (Required) Unique name of the Azure destination.
+
+* `key_vault_uri` - (Optional) URI of an existing Azure Key Vault instance.
+  Can be omitted and directly provided to Vault using the `KEY_VAULT_URI` environment
+  variable.
+
+* `tenant_id` - (Optional) JSON-encoded credentials to use to connect to Azure.
+  Can be omitted and directly provided to Vault using the `AZURE_TENANT_ID` environment
+  variable.
```
Comment (maxcoulombe) [inline website/docs/r/secrets_sync_azure_destination.html.md:55]: I think this is just an account-specific uuid, not JSON-encoded data.

File: website/docs/r/secrets_sync_association.html.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+~> **Important** All data provided in the resource configuration will be
+written in cleartext to state and plan files generated by Terraform, and
+will appear in the console output when Terraform runs. Protect these
+artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+For more information on associations, please refer to the Vault
+[documentation](https://developer.hashicorp.com/vault/docs/sync#associations).
```
Comment (maxcoulombe) [inline website/docs/r/secrets_sync_association.html.md:22]: Would be great to use mdx to have a reusable snippet for things like this :( Maybe there is a way with plain MD to do it so we'd remove some duplication!

**Review by maxcoulombe — APPROVED** [review]
Nice! Couple of small questions or minor comments but the bulk looks super good.

File: website/docs/r/secrets_sync_association.html.md (hunk 1 above)
Comment (vinay-gopalan) [reply website/docs/r/secrets_sync_association.html.md:51]: Updated!

File: website/docs/r/secrets_sync_aws_destination.html.md (hunk 2 above)
Comment (vinay-gopalan) [reply website/docs/r/secrets_sync_aws_destination.html.md:37]: Added customizable fields to the documentation for all destinations

File: website/docs/r/secrets_sync_azure_destination.html.md (hunk 3 above)
Comment (vinay-gopalan) [reply website/docs/r/secrets_sync_azure_destination.html.md:56]: Fixed

File: website/docs/r/secrets_sync_association.html.md (hunk 4 above)
Comment (vinay-gopalan) [reply website/docs/r/secrets_sync_association.html.md:22]: Ah good point. Might port some of our documentation over to `.mdx` in the future, I believe the new TF Framework generates documentation in that format. I can take a look if this is possible for `.md` files for the time being and can update the docs in a follow-up if I find a better way 👍🏼

**Pull request merged**